ggml/src/ggml-metal/ggml-metal.m

```diff
@@ -44,9 +44,10 @@
 // note: assumes single GPU device - the default one
 // TODO: support multiple GPU devices
 static struct ggml_backend_metal_device_context {
-    id<MTLDevice>  mtl_device;
-    int            mtl_device_ref_count;
-    id<MTLLibrary> mtl_library;
+    id<MTLDevice>       mtl_device;
+    int                 mtl_device_ref_count;
+    id<MTLCommandQueue> mtl_queue;
+    id<MTLLibrary>      mtl_library;
```

Comment:
At some point during the refactoring, decided to move the MTLQueue from the backend context to the device context. I.e. a single queue is now shared by all backend contexts.

Not sure if this is better. Maybe will move it back to the backend context and create separate queues for each synchronous `get_tensor`, `set_tensor` and `memset_tensor` operations.

Reply:
The intention is that the `ggml_backend` object represents a queue or stream. In practice, it probably doesn't matter too much if it is shared for Apple Mx iGPUs, but it would prevent some use cases such as running a computation in one backend and a memory transfer in another simultaneously.